### PR TITLE
Chore: Remove redundant code in prometheus datasource

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

Removing the redundant code in the Prometheus datasource.
Thread about this for more context https://raintank-corp.slack.com/archives/C02UC0KCSF4/p1696521755012539

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/datasource.ts [hunk 1]
```diff
@@ -483,10 +474,6 @@ export class PrometheusDatasource
     return processedTargets;
   }
 
-  intepolateStringHelper = (query: PromQuery): string => {
-    return this.interpolateString(query.expr);
-  };
-
```
Comment (itsmylife) [inline public/app/plugins/datasource/prometheus/datasource.ts:489]: @gtk-grafana I think that was used for testing and then forgotten. It wasn't being used anywhere so I removed it.

File: public/app/plugins/datasource/prometheus/datasource.ts [hunk 2]
```diff
@@ -97,7 +93,6 @@ export class PrometheusDatasource
   access: 'direct' | 'proxy';
   basicAuth: any;
   withCredentials: any;
-  metricsNameCache = new LRUCache<string, string[]>({ max: 10 });
```
Comment (itsmylife) [inline public/app/plugins/datasource/prometheus/datasource.ts:100]: @gtk-grafana this wasn't in use so I removed it.

Comment (gtk-grafana) [reply public/app/plugins/datasource/prometheus/datasource.ts:100]: Yeah good, call there should also be a `labelsCache` and `labelValuesCache` which can be removed in the language_provider.ts as well

**Review by bohandley — APPROVED** [review]

**Comment by itsmylife** [comment]
e2e tests are always failing

**Pull request closed without merging**